core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -427,19 +427,22 @@ private SslData createSslData(
       TrustManagerFactory tmf = TrustManagerFactory.getInstance("X.509");
       tmf.init(trustedKeyStore);
       SSLContext sslContext;
-      if (enableIamAuth) {
-        try {
-          sslContext = SSLContext.getInstance("TLSv1.3");
-        } catch (NoSuchAlgorithmException ex) {
+
+      try {
+        sslContext = SSLContext.getInstance("TLSv1.3");
+      } catch (NoSuchAlgorithmException ex) {
+        if (enableIamAuth) {
           throw new RuntimeException(
               String.format(
-                  "[%s] Unable to create a SSLContext for the Cloud SQL instance.", connectionName)
+                  "[%s] Unable to create a SSLContext for the Cloud SQL instance.",
+                  connectionName)
                   + " TLSv1.3 is not supported for your Java version and is required to connect"
                   + " using IAM authentication",
               ex);
+        } else {
+          logger.warning("TLSv1.3 is not supported for your Java version, fallback to TLSv1.2");
```

Comment:
Quick check - will this get logged too much? It's at the warning level and seems like it'll get called each refresh for each instance?

Reply:
Maybe I should demote it to info? I think it's worth logging, but also don't want to clutter the logs.

Reply:
I think warning is the right level, just concerned about the frequency. 

One a refresh is maybe an acceptable amount.

Alternatively, we could move this into a helper function that only calls the warning on the first invocation, and maybe call it when the CloudSQLinstance or socketfactory is initialized.

Reply:
I think once per refresh is fine.

Reply:
if later we decide the logs are too cluttered, we can investigate moving it into a helper function.